feat: add worktree feature with tests

racgoo opened the pull request:
## Summary
- Add 'worktree' feature with tests.

## Changes
- Implement worktree-like methods in 'Repository'
- Add internal worktree features
- Add test code

Closes #172

seokju-na commented on src/worktree.rs:
```diff
@@ -0,0 +1,435 @@
+impl From<WorktreePruneOptions> for git2::WorktreePruneOptions {
+  fn from(value: WorktreePruneOptions) -> Self {
+    let mut git2_worktree_prune_options = git2::WorktreePruneOptions::new();
+    value.valid.map(|_valid| git2_worktree_prune_options.valid(_valid));
```
```suggestion
    if let Some(valid) = value.valid {
      git2_worktree_prune_options.valid(valod);
    }
```
`.map` is is not a suitable operator to perform mutable operations. Can we change this line?

racgoo commented on src/worktree.rs:
```diff
@@ -0,0 +1,435 @@
+  /// @param {Repository} repo - Repository to open worktree from.
+  /// @returns Worktree instance.
+  /// @throws Throws error if the repository is not a worktree or if opening fails.
+  pub fn open_from_repository(repo: Reference<Repository>, env: Env) -> crate::Result<Worktree> {
```
Sorry for the late reply. spent some time thinking through the async design here.

My current thinking is to use `AsyncTask` only when the operation can be
expressed in terms of pure Rust values (e.g. String, PathBuf, primitive
types) that can safely be moved to a worker thread.

In cases like this one, a real `AsyncTask` is not possible because
napi-managed objects cannot be moved to worker threads.

One alternative is a `fake-async` Promise-based API (sync execution +
immediate resolve), which would require upgrading napi-rs to >= 3.7.1.

For reference, the implementation would look like this
```rust
pub fn open_worktree_from_repository(repo: &Repository, env: Env) -> crate::Result<PromiseRaw<'_,Worktree>> {
  let git2_worktree = git2::Worktree::open_from_repository(&repo.inner);
  match git2_worktree {
    Ok(worktree) => {
      Ok(PromiseRaw::resolve(&env, Worktree { inner: worktree })?)
    },
    Err(error) => {
      Ok(PromiseRaw::reject(&env, error.to_string())?)
    }
  }
}
```
Does this approach sound reasonable to you?

racgoo commented:
@seokju-na  I spent a few days thinking about how to make this async safely.

Based on libgit2's threading rules
(https://github.com/libgit2/libgit2/blob/68a3c0b1244687ce07c3d8605e875485d86b05ff/docs/threading.md),
I couldn’t find a safe way to run this operation on a worker thread.
git2-rs / napi-managed objects cannot be moved across threads, so a real `AsyncTask` does not seem possible here.

As a result, I implemented this as a Promise-based API that executes synchronously
and resolves immediately on the main thread.
To keep the implementation simple and consistent, I also added a small helper macro
to create resolved / rejected promises.

Happy to adjust this if you think another approach would fit better.
